### PR TITLE
Convert preinc/predec in 3rd part of for statement to postinc/postdec

### LINKED ISSUE #16138: do not preincrement unnecessarily
```
libr/anal/p/anal_riscv.c:		//	for (i = 0; i < args.num; ++i) {
libr/anal/p/anal_rsp.c:	for (i = 0; i < r_instr.noperands; ++i) {
libr/anal/xrefs.c:		for (i = 0; i < bb->ninstr; ++i) {
libr/asm/arch/arc/gnu/arc-opc.c:	  for (i = 0; i < (int)ELEMENTS_IN (arc_operands_a4); ++i) {
libr/asm/arch/arc/gnu/arc-opc.c:	  for (i = 0; i < (int)ELEMENTS_IN (arc_operands_ac); ++i) {
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; i < ARRAY_SIZE (self->fields) && self->fields[i] != FLD_NIL; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; aarch64_pstatefields[i].name != NULL; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; sysins_ops[i].name != NULL; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; i < AARCH64_MAX_QLF_SEQ_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; i < AARCH64_MAX_QLF_SEQ_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:	   && i < AARCH64_MAX_QLF_SEQ_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:  for (i = 0, num_printed = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-dis.c:      for (i = 1; i < num_conds && inst->cond->names[i]; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0, saved_i = -1; i < AARCH64_MAX_QLF_SEQ_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i, ++qualifier)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0; i < AARCH64_MAX_QLF_SEQ_NUM; ++i, ++qualifiers_list)
libr/asm/arch/arm/aarch64/aarch64-opc.c:      for (i = 0; i < nops; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:    for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:      for (i = 1; i < num_conds && opnd->cond->names[i]; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:      for (i = 0; aarch64_sys_regs[i].name; ++i)
libr/asm/arch/arm/aarch64/aarch64-opc.c:      for (i = 0; aarch64_pstatefields[i].name; ++i)
libr/asm/arch/arm/aarch64/aarch64.h:  for (i = 0; i < AARCH64_MAX_OPND_NUM; ++i)
libr/asm/arch/hppa/gnu/hppa-dis.c:  for (i = 0; i < NUMOPCODES; ++i)
libr/asm/arch/lanai/gnu/lanai-dis.c:  for (i = 0; i < NUMOPCODES; ++i)
libr/asm/arch/lanai/gnu/lanai-dis.c:  for (i = 0; i < NUMOPCODES; ++i)
libr/asm/arch/lanai/gnu/lanai-dis.c:  for (i = 0; i < 32; ++i)
libr/asm/arch/lanai/gnu/lanai-dis.c:  for (i = 0; i < 32; ++i)
libr/asm/arch/mcore/mcore.c:	for (i = 0; i < op_ptr->n_args; ++i) {
libr/asm/arch/mcore/mcore.c:	for (i = 0; add > 0 && i < instr->n_args && add < bufsize; ++i) {
libr/asm/arch/nios/gnu/nios2-dis.c:  for (i = 0; i <= OP_MASK_OP; ++i) {
libr/asm/arch/nios/gnu/nios2-dis.c:  for (i = 0; i <= OP_MASK_OP; ++i)
libr/asm/arch/nios/gnu/nios2-dis.c:  for (i = 0; i <= OP_MASK_OP; ++i)
libr/asm/arch/ppc/libps/libps.c:	for (i = 0; add > 0 && i < instr->n && add < bufsize; ++i) {
libr/asm/arch/ppc/libvle/vle.c:	for (i = 0; i < size; ++i) {
libr/asm/arch/ppc/libvle/vle.c:	for (i = 0; i < size; ++i) {
libr/asm/arch/ppc/libvle/vle.c:	for (i = 0; i < size; ++i) {
libr/asm/arch/ppc/libvle/vle.c:	for (i = 0; add > 0 && i < instr->n && add < bufsize; ++i) {
libr/asm/arch/sparc/gnu/sparc-dis.c:  for (i = 0; i < 32; ++i)
libr/asm/arch/sparc/gnu/sparc-dis.c:  for (i = 0; i < 32; ++i)
libr/asm/arch/sparc/gnu/sparc-dis.c:    for (i = 0; i < HASH_SIZE; ++i)
libr/asm/arch/sparc/gnu/sparc-dis.c:      for (i = 0; i < sparc_num_opcodes; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < dec_insn.code->nr_operands; ++i) {
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0, pop = tricore_opcodes; i < NUMOPCS; ++i, ++pop)
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0, ppop = pcp_opcodes; i < NUMPCPOPCS; ++i, ++ppop)
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0, psfr = tricore_sfrs; i < NUMSFRS; ++i, ++psfr)
libr/asm/arch/tricore/gnu/tricore-dis.c:  for (i = 0; i < insn->nr_operands; ++i)
libr/asm/arch/tricore/gnu/tricore-dis.c:      for (idx = 0; idx < pop->nr_operands; ++idx)
libr/asm/arch/tricore/gnu/tricore-dis.c:      for (idx = 0; idx < pop->nr_operands; ++idx)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:	  for (i = 0; i < o_operand_count; ++i)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:  for (i = 0; i < sizeof (action) / sizeof (*action); ++i)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:  for (i = 0; r; ++i, r = r->next)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:  for (i = 0; i < ebb_table->action_count; ++i)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:	  for (i = 1; i < constraint->action_count; ++i)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:  for (++i; i < src_count; ++i)
libr/asm/arch/xtensa/gnu/elf32-xtensa.c:      for (i = 0; i < sec->reloc_count; ++i)
libr/asm/p/asm_rsp.c:	for (i = 0; i < r_instr.noperands; ++i) {
libr/bin/format/elf/elf.c:	for (i = 0; i < bin->ehdr.e_shnum; ++i) {
libr/bin/format/elf/elf.c:	for (i = 0; i < bin->ehdr.e_phnum; ++i) {
libr/bin/format/elf/elf.c:	for (i = 0; i < bin->ehdr.e_phnum; ++i) {
libr/bin/format/elf/elf.c:	for (i = 0; i < bin->ehdr.e_phnum; ++i) {
libr/bin/format/elf/elf.c:	for (i = 0; i < bin->ehdr.e_phnum; ++i) {
libr/bin/format/mach0/mach0.c:	for (i = 0; i < bin->nsegs; ++i) {
libr/bin/format/pe/pe.c:	for (i = 0; *curAddr < begAddr + string->wLength; ++i, *curAddr += sizeof (ut16)) {
libr/bin/mangling/cxx/cp-demangle.c:  for (i = 0; i < indent; ++i)
libr/bin/mangling/cxx/cp-demangle.c:	for (i = 0; i < len; ++i)
libr/bin/mangling/cxx/cp-demangle.c:      for (i = optind; i < argc; ++i)
libr/bin/p/bin_dyldcache.c:	for (i = 0; i < hdr->mappingCount; ++i) {
libr/bin/p/bin_dyldcache.c:	for (i = 0; i < cache->hdr->mappingCount; ++i) {
libr/bin/p/bin_dyldcache.c:	for (i = 0; i < cache->hdr->mappingCount; ++i) {
libr/cons/2048.c:		for (i = 0; i < 4; ++i) {
libr/cons/rgb.c:	for (i = 16; i < 256; ++i) {
libr/core/agraph.c:		for (i = 1; i < diff_layer; ++i) {
libr/core/agraph.c:	for (i = 0; i < g->n_layers; ++i) {
libr/core/agraph.c:		for (i = 0; i < g->n_layers; ++i) {
libr/core/agraph.c:		for (i = aa->pos_in_layer; i < ab->pos_in_layer; ++i) {
libr/core/agraph.c:	for (i = 0; i < g->n_layers; ++i) {
libr/core/agraph.c:	for (i = 0; i < g->n_layers; ++i) {
libr/core/agraph.c:	for (i = 0; i < n_classes; ++i) {
libr/core/agraph.c:	for (i = 0; i < n_classes; ++i) {
libr/core/agraph.c:	for (i = t + 1; i < r; ++i) {
libr/core/agraph.c:		for (i = 0; i < g->n_layers; ++i) {
libr/core/agraph.c:		for (i = 0; i < g->n_layers; ++i) {
libr/core/agraph.c:	for (i = 0; i < g->n_layers; ++i) {
libr/core/casm.c:		for (ii = 0; ii < asmlen; ++ii) {
libr/core/cmd.c:	for (i = 0; i < s_len; ++i) {
libr/core/cmd.c:	for (i = 0; i < a->argc; ++i) {
libr/core/cmd.c:		for (i = 0; i < n_children; ++i) {
libr/core/cmd.c:		for (i = 0; i < n_children; ++i) {
libr/core/cmd.c:	for (i = 0; i < a->argc; ++i) {
libr/core/cmd.c:		for (i = 0; i < n_children; ++i) {
libr/core/cmd.c:		for (i = 0; i < pr_args->argc; ++i) {
libr/core/cmd.c:	for (i = 0; i < rep; ++i) {
libr/core/cmd.c:	for (i = 0; i < n_args; ++i) {
libr/core/cmd.c:	for (i = 0; i < a->argc; ++i) {
libr/core/cmd.c:	for (i = 0; i < a->argc; ++i) {
libr/core/cmd.c:	for (i = 0; i < child_count; ++i) {
libr/core/cmd_anal.c:			for (i = idx, j = 0; i < core->blocksize && j < 3; ++i, ++j) {
libr/core/cmd_seek.c:			for (i = 0; i < lsz; ++i) {
libr/core/core.c:		for (i = 0; i < len - 1; ++i) {
libr/core/core.c:	for (i = 0; i < obj->n_subcmds; ++i) {
libr/core/core.c:	for (i = 0; i < parent->n_subcmds; ++i) {
libr/core/disasm.c:			for (i = 0; i < len; ++i) {
libr/core/panels.c:			for (i = 4, j = 4; i < len; ++i,++j) {
libr/core/pseudo.c:	for (ctx.linebegin = in; in < end; ++in) {
libr/core/visual.c:			for (i = 0; i < numinstrs; ++i) {
libr/core/visual.c:	for (i = 0; i < numinstrs; ++i) {
libr/core/visual.c:					for (i = 4, j = 4; i < len; ++i,++j) {
libr/crypto/p/crypto_cps2.c:	for (i = 0; i < 96; ++i) {
libr/crypto/p/crypto_cps2.c:	for (i = 0; i < 96; ++i) {
libr/crypto/p/crypto_cps2.c:	for (i = 0; i < 64; ++i) {
libr/crypto/p/crypto_cps2.c:	for (i = 0; i < 6; ++i) {
libr/crypto/p/crypto_cps2.c:		for (i = 0; i < 256; ++i) {
libr/crypto/p/crypto_cps2.c:		for (i = 0; i < 64; ++i) {
libr/crypto/p/crypto_cps2.c:	for (i = 0; i < 0x10000; ++i) {
libr/crypto/p/crypto_des.c:	for (i = 0; i < 16; ++i) {
libr/flag/flag.c:		for (i = 0; i < n_spaces; ++i) {
libr/hash/fletcher.c:		for (i = 0; i < 5802; ++i) {
libr/hash/fletcher.c:	for (i = 0; i < len; ++i) {
libr/main/radiff2.c:	for (i = 0; i < len; ++i) {
libr/reg/arena.c:	for (i = 0; i < R_REG_TYPE_LAST; ++i) {
libr/reg/reg.c:	for (i = 0; i < R_REG_TYPE_LAST; ++i) {
libr/reg/reg.c:		for (i = 0; i < R_REG_TYPE_LAST; ++i) {
libr/reg/reg.c:	for (i = 0; i < R_REG_TYPE_LAST; ++i) {
libr/reg/reg.c:		for (i = 0; i < R_REG_TYPE_LAST; ++i) {
libr/util/asn1.c:			for (i8 = 0; i8 < length8; ++i8) {
libr/util/asn1.c:			for (i = 0; next >= buffer && next < end && i < count; ++i) {
libr/util/asn1.c:	for (i = 0; i < object->length && p < end; ++i) {
libr/util/asn1.c:			for (i = 0; i < object->list.length; ++i) {
libr/util/asn1.c:			for (i = 0; i < object->list.length; ++i) {
libr/util/asn1.c:		for (i = 0; i < object->list.length; ++i) {
libr/util/astr.c:	for (i = 1, j = 0; i < length && j < size; ++i) {
libr/util/astr.c:	for (i = 0, j = 0; i < length && j < size; ++i, j += 3) {
libr/util/astr.c:	for (i = 0, j = 0, k = 48; i < length && j < size && k < size; ++i, j += 3, k++) {
libr/util/buf.c:	for (i = 0; i < n; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < crls->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < crls->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < ecac->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < ecac->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < dai->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < dai->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < attributes->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < ss->length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < object->list.length; ++i) {
libr/util/pkcs7.c:	for (i = 0; i < crl->length; ++i) {
libr/util/pkcs7.c:	for (i = 0; i < si->authenticatedAttributes.length; ++i) {
libr/util/pkcs7.c:	for (i = 0; i < si->unauthenticatedAttributes.length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < container->signedData.digestAlgorithms.length; ++i) {
libr/util/pkcs7.c:	for (i = 0; i < container->signedData.certificates.length; ++i) {
libr/util/pkcs7.c:	for (i = 0; i < container->signedData.crls.length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < container->signedData.signerinfos.length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < si->authenticatedAttributes.length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < si->unauthenticatedAttributes.length; ++i) {
libr/util/pkcs7.c:			for (i = 0; i < container->signedData.digestAlgorithms.length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < container->signedData.certificates.length; ++i) {
libr/util/pkcs7.c:		for (i = 0; i < container->signedData.crls.length; ++i) {
libr/util/pkcs7.c:			for (i = 0; i < container->signedData.signerinfos.length; ++i) {
libr/util/str.c:	for (i = 1; i < sz; ++i) {
libr/util/x509.c:		for (i = 0; i < object->list.length; ++i) {
libr/util/x509.c:	for (i = 0; i < object->list.length; ++i) {
libr/util/x509.c:		for (i = 0; i < object->list.objects[4]->list.length; ++i) {
libr/util/x509.c:		for (i = 0; i < name->length; ++i) {
libr/util/x509.c:		for (i = 0; i < ex->length; ++i) {
libr/util/x509.c:			for (i = 0; i < crl->length; ++i) {
libr/util/x509.c:	for (i = 0; i < name->length; ++i) {
libr/util/x509.c:	for (i = 0; i < exts->length; ++i) {
libr/util/x509.c:	for (i = 0; i < name->length; ++i) {
libr/util/x509.c:		for (i = 0; i < exts->length; ++i) {
libr/util/x509.c:		for (i = 0; i < crl->length; ++i) {
shlr/grub/partmap/acorn.c:  for (i = 0; i != 0x1ff; ++i)
shlr/grub/partmap/acorn.c:  for (i = 0; i != LINUX_MAP_ENTRIES; ++i)
shlr/sdb/src/ht_inc.c:	for (i = 0; i < ht->size; ++i) {
shlr/sdb/src/sdb.c:	for (i = 0; i < s->ht->size; ++i) {
shlr/sdb/src/sdb.c:	for (i = 0; i < s->ht->size; ++i) {
shlr/spp/main.c:	for( i = 0; procs[i]; ++i ) {
shlr/tcc/tccpp.c:		for (i = -2; i < n; ++i) {
shlr/www/p/vendors/dagre.core.js:  for (i = 0, ++vRank; vRank < wRank; ++i, ++vRank) {
shlr/www/p/vendors/dagre.core.js:  for (var i = 1; i < layering.length; ++i) {
shlr/www/p/vendors/dagre.core.js:  for (var i = 0, lastBest = 0; lastBest < 4; ++i, ++lastBest) {
shlr/www/p/vendors/dagre.core.js:        for (var i = Math.floor(mp), il = Math.ceil(mp); i <= il; ++i) {
test/unit/legacy_unit/util/test_str.c:	for (i = 0; i < len; ++i) {
test/unit/test_list.c:	for (i = 0; i < 100; ++i) {
test/unit/test_list.c:	for (i = 0; i < 50; ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (ins_tests_odd); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (exp_tests_odd); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (exp_tests_even); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (test); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (test); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (test); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (test); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (test); ++i) {
test/unit/test_list.c:	for (i = 0; i < R_ARRAY_SIZE (test); ++i) {
test/unit/test_queue.c:	for (i = 0; i < 5; ++i) {
test/unit/test_queue.c:	for (i = 0; i < 6; ++i) {
test/unit/test_skiplist.c:	for (i = 0; i < 100; ++i) {
test/unit/test_skiplist.c:	for (i = 0; i < 100; ++i) {
test/unit/test_table.c:	for (i = 0; i < 4; ++i) {
test/unit/test_unum.c:	for (i = 0; i < nitems; ++i) {
```

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

This pr fixes #16138 harder by:
1. Being more thorough.
1. Converting also predecrements.
1. Adding a test e4cbf895bbaf94dfeaf9e5e68e0a5a0ca377fd58 (not entirely sure this is a good idea + it takes a few seconds). The test is incomplete (i.e. there are false negatives) but should be sound (i.e. no false positives).

Conversion was done using the bash command:

```bash
REGEX='\(for[[:space:]]*([^;)]*;[^;)]*;\([^,)(]*,\)*[[:space:]]*\)\(++\|--\)[[:space:]]*\([[:alpha:]_][[:alnum:]_]*\)\([[:space:]]*[,)]\)' bash -c 'git grep -l "$REGEX" -- :!**/db/tools/r2 :!shlr/** :!libr/asm/arch/** :!**/mangling/cxx/cp-demangle.c | xargs -n 1 sed -i "s/$REGEX/\1\4\3\5/g"'
```

repeatedly until the test is green.

~~For the minified js, manual checking of their diffs can be done using~~

```bash
git diff --word-diff-regex=.
```

~~which does character-based diffing.~~

**Test plan**

Travis and AppVeyor are green.

**Closing issues**

See description.

### CODE REVIEW

**Comment by kazarmy** [comment]
Added `-- :^**/db/tools/r2` to the `git grep` subcommand in the bash conversion command so that it doesn't fix the test comment.

**Comment by XVilka** [comment]
By the way, instead of complicated regexps you can use  [Coccinelle](https://github.com/coccinelle/coccinelle), see their [documentation](http://coccinelle.lip6.fr/documentation.php) for more details. There is a good [introductory article on how to create semantic patches](https://home.regit.org/technical-articles/coccinelle-for-the-newbie/). Coccinelle used for the Linux kernel sources refactoring, so it's stable and featureful enough.
See the example of this semantic patch:
```patch
@@
expression lock, flags;
expression urb;
@@

 spin_lock_irqsave(lock, flags);
 <...
- usb_submit_urb(urb)
+ usb_submit_urb(urb, GFP_ATOMIC)
 ...>
 spin_unlock_irqrestore(lock, flags);

@@
expression urb;
@@

- usb_submit_urb(urb)
+ usb_submit_urb(urb, GFP_KERNEL)
```

File: libr/asm/arch/arc/gnu/arc-opc.c [hunk 1]
```diff
@@ -4504,7 +4504,7 @@ arc_opcode_lookup_suffix (const struct arc_operand *type, int value)
     }
 
   /* ??? This is a little slow and can be speeded up.  */
-    for (v = arc_suffixes, end = arc_suffixes + arc_suffixes_count; v < end; ++v) {
+    for (v = arc_suffixes, end = arc_suffixes + arc_suffixes_count; v < end; v++) {
```
Comment (radare) [inline libr/asm/arch/arc/gnu/arc-opc.c]: I didnt updated this one because its from gnu

File: shlr/sdb/src/sdb.c [hunk 2]
```diff
@@ -791,7 +791,7 @@ SDB_API bool sdb_foreach(Sdb* s, SdbForeachCallback cb, void *user) {
 	}
 
 	ut32 i;
-	for (i = 0; i < s->ht->size; ++i) {
+	for (i = 0; i < s->ht->size; i++) {
```
Comment (radare) [inline shlr/sdb/src/sdb.c]: This goes into the sdb repo. Not the r2 one

File: shlr/www/p/vendors/dagre.core.js [hunk 3]
```diff
@@ -1038,7 +1038,7 @@ function normalizeEdge(g, e) {
   g.removeEdge(e);
 
   var dummy, attrs, i;
-  for (i = 0, ++vRank; vRank < wRank; ++i, ++vRank) {
+  for (i = 0, ++vRank; vRank < wRank; i++, vRank++) {
```
Comment (radare) [inline shlr/www/p/vendors/dagre.core.js]: This is autogenerated code

File: libr/asm/arch/arc/gnu/arc-opc.c (hunk 1 above)
Comment (kazarmy) [reply libr/asm/arch/arc/gnu/arc-opc.c]: ok 2258c7faa95fc2c19e1fee59d3d1ffe8cc8297cd

File: shlr/sdb/src/sdb.c (hunk 2 above)
Comment (kazarmy) [reply shlr/sdb/src/sdb.c]: ok excluded 2258c7faa95fc2c19e1fee59d3d1ffe8cc8297cd

File: shlr/www/p/vendors/dagre.core.js (hunk 3 above)
Comment (kazarmy) [reply shlr/www/p/vendors/dagre.core.js]: ok 2258c7faa95fc2c19e1fee59d3d1ffe8cc8297cd

**Comment by kazarmy** [comment]
> By the way, instead of complicated regexps you can use Coccinelle, ...

Thanks for the suggestion! But I were to use Coccinelle:

1. I would need to learn it.
1. Code what is needed + work around any bugs.
1. Integrate it with r2r.

That, for me at least, would require tens of hours of work at the minimum, and I would still need to get it past @radare which is hardly a given, since it's going to be yet another dependency.

Besides, I think the regexp looks complicated because it lacks spaces. If I were to write it this way:

```
for  [[:space:]]*  (  [^;)]*  ;  [^;)]*  ;  \([^,)(]*,\)*  [[:space:]]*  \(++\|--\)  [[:space:]]*  [[:alpha:]_][[:alnum:]_]*  [[:space:]]*  [,)]
```

I think it becomes a lot less intimidating.

That said, this is probably stretching the limits of appropriate regex usage. They may look like complicated beasts, but automata theory will tell you that they aren't that powerful.

**Pull request merged**